### PR TITLE
commons #109 add toMap method for ConfigurationImplicits

### LINKED ISSUE #109: add toMap method in ConfigurationImplicits
this method will convert the configuration into Scala map

### PR DESCRIPTION
**Pull request by cerveada**
fixes #109

### CODE REVIEW

**Review by dk1844 — COMMENTED** [review]
LGTM (just read the code)

**Comment by wajda** [comment]
1. According to SemVer principles this change is worth incrementing the MINOR version component.
2. Shouldn't this new method be also mentioned in the README?

**Comment by cerveada** [comment]
> 1. According to SemVer principles this change is worth incrementing the MINOR version component.

ok

> 2. Shouldn't this new method be also mentioned in the README?

it is

**Review by wajda — APPROVED** [review]
LGTM

**Comment by cerveada** [comment]
@wajda @dk1844 Sorry for changing things after code review, but I find out why `TypeTag` wasn't working for me before. So I change `ClassTag` to `TypeTag` and added support for `Array[String]` as well.

File: src/test/scala/za/co/absa/commons/config/ConfigurationImplicitsSpec.scala [hunk 1]
```diff
@@ -225,4 +225,59 @@ class ConfigurationImplicitsSpec extends AnyFlatSpec with Matchers {
...
+  it should "implement convert config to map of Integers" in {
+    val configuration = new MapConfiguration(Map("foo" -> "1", "bar" -> 2).asJava)
+    val map = configuration.toMap[Int]
+    map.apply("foo") should be(1)
+    map.apply("bar") should be(2)
+  }
+
+  it should "implement convert config to map of Longs" in {
+    val configuration = new MapConfiguration(Map("foo" -> "1", "bar" -> 2).asJava)
+    val map = configuration.toMap[Int]
```
Comment (miroslavpojer) [inline src/test/scala/za/co/absa/commons/config/ConfigurationImplicitsSpec.scala:247]: I would expect type Long in this test.
```suggestion
    val map = configuration.toMap[Long]
```

Comment (miroslavpojer) [reply src/test/scala/za/co/absa/commons/config/ConfigurationImplicitsSpec.scala]: Unit tests executed.

**Review by wajda — APPROVED** [review]

File: src/test/scala/za/co/absa/commons/config/ConfigurationImplicitsSpec.scala (hunk 1 above)
Comment (miroslavpojer) [reply src/test/scala/za/co/absa/commons/config/ConfigurationImplicitsSpec.scala]: Sure, This is only note that related unit tests have been executed.

Comment (cerveada) [reply src/test/scala/za/co/absa/commons/config/ConfigurationImplicitsSpec.scala]: ok, do you see anything else that needs to be fixed? If not please approve the PR.

**Review by miroslavpojer — APPROVED** [review]

**Pull request merged**